src/ompl/geometric/planners/rrt/RRTstar.h

```diff
@@ -173,17 +190,57 @@ namespace ompl
                 return pruneStatesThreshold_;
             }
 
+            /** \brief Use a k-nearest search for rewiring instead of a r-disc search. */
+            void setKNearest(bool useKNearest)
+            {
+                useKNearest_ = useKNearest;
+            }
+
+            /** \brief Get the state of using a k-nearest search for rewiring. */
+            bool getKNearest() const
+            {
+                return useKNearest_;
+            }
+
+            /** \brief Use \e Informed \e RRT*.
+            This means that once a problem is found, the search is focused only to the subproblem that could contain a better solution.
+            Currently only implemented for problems with a single goal that are seeking to minimize path length in
+            R^n (i.e., RealVectorStateSpace), SE(2) (i.e., SE2StateSpace), or SE(3) (i.e., SE3StateSpace).
+            @par J D. Gammell, S. S. Srinivasa, T. D. Barfoot, "Informed RRT*: Optimal Sampling-based
+            Path Planning Focused via Direct Sampling of an Admissible Ellipsoidal Heuristic."
+            IROS 2014. DOI: <a href="http://dx.doi.org/10.1109/IROS.2014.6942976">10.1109/IROS.2014.6942976</a>.
+            <a href="http://www.youtube.com/watch?v=d7dX5MvDYTc">Illustration video</a>.
+            <a href="http://www.youtube.com/watch?v=nsl-5MZfwu4">Short description video</a>. */
+            void setInformedSampling(bool informedSampling);
+
+            /** \brief Get the state of informed sampling */
+            bool getInformedSampling() const
+            {
+                return useInformedSampling_;
+            }
+
+            /** \brief Set the number of attempts to make while performing informed sampling */
+            void setInformedSamplingAttempts(unsigned int numAttempts)
+            {
+                numInfAttempts_ = numAttempts;
+            }
+
+            /** \brief Get the number of attempts to make while performing informed sampling */
+            unsigned int getInformedSamplingAttempts() const
+            {
+                return numInfAttempts_ ;
+            }
+
             virtual void setup();
 
-            ///////////////////////////////////////
-            // Planner progress property functions
-            std::string getIterationCount() const
+            unsigned int numIterations() const
             {
-                return boost::lexical_cast<std::string>(iterations_);
+                return iterations_;
             }
-            std::string getBestCost() const
+
+            ompl::base::Cost bestCost() const
             {
-                return boost::lexical_cast<std::string>(bestCost_);
+                return bestCost_;
             }
```

Comment:
This sounds good, but do you need this as a PlannerProgressProperty? Properties must return string, so maybe duplicate the function? Add a suffix "Property" to the function name, leave the private version as returning string and return the actual cost / nr iterations in public functions?